src/java/com/google/cloud/bigquery/dwhassessment/hooks/MigrationAssessmentLoggingHook.java

```diff
@@ -42,87 +28,12 @@ public class MigrationAssessmentLoggingHook implements ExecuteWithHookContext {
-    QueryPlan plan = hookContext.getQueryPlan();
-
-    LOG.info("Received hook notification for: {}", plan.getQueryId());
-
-    // Make a copy so that we do not modify hookContext conf.
-    HiveConf conf = new HiveConf(hookContext.getConf());
-    List<ExecDriver> mrTasks = Utilities.getMRTasks(plan.getRootTasks());
-    List<TezTask> tezTasks = Utilities.getTezTasks(plan.getRootTasks());
-    ExecutionMode executionMode = getExecutionMode(mrTasks, tezTasks);
-
-    GenericRecord queryEvent =
-        new GenericRecordBuilder(QUERY_EVENT_SCHEMA)
-            .set("QueryId", plan.getQueryId())
-            .set("QueryText", plan.getQueryStr())
-            .set("EventType", hookContext.getHookType().name())
-            .set("Timestamp", plan.getQueryStartTime())
-            .set("User", getUser(hookContext))
-            .set("RequestUser", getRequestUser(hookContext))
-            .set("ExecutionMode", executionMode.name())
-            .set("Queue", getQueueName(executionMode, conf))
-            .set("TablesRead", getTablesFromEntitySet(plan.getInputs()))
-            .set("TablesWritten", getTablesFromEntitySet(plan.getOutputs()))
-            .build();
-    records.add(queryEvent);
-    LOG.info("Processed record: {}", queryEvent);
-  }
-
-  private List<String> getTablesFromEntitySet(Set<? extends Entity> entities) {
-    List<String> tableNames = new ArrayList<>();
-    for (Entity entity : entities) {
-      if (entity.getType() == TABLE || entity.getType() == PARTITION) {
-        tableNames.add(entity.getTable().getCompleteName());
-      }
-    }
-    return tableNames;
-  }
-
-  private String getUser(HookContext hookContext) {
-    return hookContext.getUgi().getShortUserName();
-  }
-
-  private String getRequestUser(HookContext hookContext) {
-    String requestUser = hookContext.getUserName();
-    return requestUser == null ? hookContext.getUgi().getUserName() : requestUser;
-  }
-
-  private ExecutionMode getExecutionMode(List<ExecDriver> mrTasks, List<TezTask> tezTasks) {
-    if (tezTasks.size() > 0) {
-      // Need to go in and check if any of the tasks is running in LLAP mode.
-      for (TezTask tezTask : tezTasks) {
-        if (tezTask.getWork().getLlapMode()) {
-          return ExecutionMode.LLAP;
-        }
-      }
-      return ExecutionMode.TEZ;
-    }
-
-    return mrTasks.size() > 0 ? ExecutionMode.MR : ExecutionMode.NONE;
-  }
-
-  private String getQueueName(ExecutionMode mode, HiveConf conf) {
-    switch (mode) {
-      case LLAP:
-        return conf.get(HiveConf.ConfVars.LLAP_DAEMON_QUEUE_NAME.varname);
-      case MR:
-        return "TODO_MAPREDUCE";
-      case TEZ:
-        return "TODO_TEZ";
-      case NONE:
-      default:
-        return null;
+    try {
+      EventLogger logger = EventLogger.getInstance(hookContext.getConf(), new SystemClock());
+      logger.handle(hookContext);
+    } catch (Exception e) {
+      LOG.error("Got exceptoin while processing event: ", e);
```

Comment:
1. Please fix typo in `exceptoin`.
2. The colon at the end of the error message is unnecessary and can be removed.

Reply:
Done